modules/mod_ginger_sword/mod_ginger_sword.erl

```diff
@@ -0,0 +1,232 @@
+-module(mod_ginger_sword).
+-author("Driebit <tech@driebit.nl>").
+
+-mod_title("SWORD support").
+-mod_description("Upload resources to a SWORD endpoint").
+-mod_prio(500).
+-mod_depends([mod_ginger_rdf]).
+
+-export([
+         observe_rsc_update_done/2,
+         dispatch_task/3,
+         cancel_tasks/1
+        ]).
+
+-include_lib("zotonic.hrl").
+
+sword_url(Context) ->
+    z_convert:to_list(m_config:get_value(mod_ginger_sword, sword_url, Context)).
+
+slug_prefix(Context) ->
+    z_convert:to_list(m_config:get_value(mod_ginger_sword, slug_prefix, Context)).
+
+%% Functions that format the data for transport
+
+format_slug(RscId, Context) ->
+    case slug_prefix(Context) of
+        undefined ->
+            z_convert:to_list(RscId);
+        Prefix when is_binary(Prefix) ->
+            binary_to_list(Prefix) ++ "-" ++ z_convert:to_list(RscId);
+        Prefix when is_list(Prefix) ->
+            Prefix ++ "-" ++ z_convert:to_list(RscId)
+    end.
+
+format_rsc_uri(RscId, Context) ->
+    sword_url(Context) ++ "/" ++ format_slug(RscId, Context).
+
+render_rdf_body(RscId, Context) ->
+    AC = z_acl:sudo(Context), % We want to publish all properties even if hidden
+    case m_rsc:exists(RscId, Context) of
+        true ->
+            RdfResource = m_rdf:to_triples(RscId, AC),
+            JsonLd = ginger_json_ld:serialize(RdfResource),
+            iolist_to_binary(mochijson2:encode(JsonLd));
+        false ->
+            <<"">>
+    end.
+
+http_request(Method, Request, _Context) ->
+    Resp = httpc:request(Method, Request, [], []),
+    {Status, _} = Resp,
+    case Status of
+        ok ->
+            Resp;
+        _ ->
+            lager:warning("Http request failed"),
+            lager:info("~p~n", [Request]),
+            lager:info("~p~n", [Resp]),
+            Resp
+    end.
```

Comment:
You can use `ginger_http_client` instead for better separation of concerns.

Reply:
Done!